src/Database/Database.php

```
@@ -535,6 +535,222 @@ public function createAttribute(string $collection, string $id, string $type, in
+    {
+        $success = true;
+
+        $this->updateAttributeMeta($collection, $id, function($attribute) use($collection, $id, $type, $size, $signed, $array, &$success) {
+            if($type !== null || $size !== null || $signed !== null || $array !== null) {
+                $type ??= $attribute->getAttribute('type');
+                $size ??= $attribute->getAttribute('size');
+                $signed ??= $attribute->getAttribute('signed');
+                $array ??= $attribute->getAttribute('array');
+                
+                switch ($type) {
+                    case self::VAR_STRING:
+                        if ($size > $this->adapter->getStringLimit()) {
+                            throw new Exception('Max size allowed for string is: ' . number_format($this->adapter->getStringLimit()));
+                        }
+                        break;
+        
+                    case self::VAR_INTEGER:
+                        $limit = ($signed) ? $this->adapter->getIntLimit() / 2 : $this->adapter->getIntLimit();
+                        if ($size > $limit) {
+                            throw new Exception('Max size allowed for int is: ' . number_format($limit));
+                        }
+                        break;
+                    case self::VAR_FLOAT:
+                    case self::VAR_BOOLEAN:
+                        break;
+                    default:
+                        throw new Exception('Unknown attribute type: ' . $type);
+                        break;
+                }
+    
+                $attribute
+                    ->setAttribute('type', $type)
+                    ->setAttribute('size', $size)
+                    ->setAttribute('signed', $signed)
+                    ->setAttribute('array', $array);
+    
+                $success = $this->adapter->updateAttribute($collection, $id, $type, $size, $signed, $array);
```

You don't need to assign this, it should throw an error

---

PDO `execute` returns boolean, so we just return that. I have seen this logic in `createAttribute` or `createAttribute`.